src/components/LogViewer/index.tsx

```diff
@@ -50,12 +51,17 @@ export type Props = PropsFromState & PropsFromDispatch;
 
 const compiledGetters = new CompiledGetterCache();
 
+function useQuery() {
+  return new URLSearchParams(useLocation().search);
```

Comment:
`new UrlSearchParams` will return a new reference each time, so I think this could cause weirdness even though useLocation() should have hook state mechanics...

I suspect you want `useEffect(() => ..., [location])` or useMemo

Reply:
I basically got this from the official react router docs https://reacttraining.com/react-router/web/example/query-parameters. I've added `useMemo` as an optimization, but I can assure you that there is no weirdness either way, unless the new search params object is passed to a child component which checks props by reference.

Hooks are just functions. If we're fine with `parseSomething(string)` in render then it's not that different for `new UrlSearchParams(string)`. `useMemo` can then be applied as [optimization but not semantic guarantee](https://reactjs.org/docs/hooks-reference.html#usememo). And I don't think `useEffect` is right here -- we just want a simple value to pass to another hook and there is no side effect.

Reply:
Yes, that's what I was worried about: `query.get("name")` is fine but passing down `query` could cause unnecessary re-renders.

Thanks for updating it!